lib/octokit/client/stats.rb

```diff
@@ -75,11 +75,26 @@ def punch_card_stats(repo, options = {})
       #
       # @param repo [Integer, String, Hash, Repository] A GitHub repository
       # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      # @return [Array<Sawyer::Resource> or nil] Magical unicorn stats or nil if unicorn haven't calculated stats yet
       def get_stats(repo, metric, options = {})
-        data = get("#{Repository.path repo}/stats/#{metric}", options)
+        get_stats_data_patiently(repo, metric, options).tap do
+          return nil if last_response.status == 202
```

Comment:
Instead of always retrying, I'd rather opt-in. I'd also like to avoid yet another configuration option. 

What if this looked like (with hypothetical `Client#get_with_retry` method):

```ruby
def get_stats(repo, metric, options = {})
  path = "#{Repository.path repo}/stats/#{metric}"
  if retry_timeout = options.delete(:retry_timeout)
    get_with_retry(path, options)
  else
    get(path, options)
  end
end
```
We could make the logic in `get_with_retry` explicitly look for a `202` since I _think_ that's the only case where we'd want to retry blindly.